ns-convert

```diff
@@ -1,184 +1,341 @@
-            if prefix is not None:
-                key = prefix + key
-            target.attrs[key] = value
-
-    @classmethod
-    def dtype_by_event(cls, event):
-        type_map = {ns.EventEntity.EVENT_TEXT  : 'a',
-                    ns.EventEntity.EVENT_CSV   : 'a',
-                    ns.EventEntity.EVENT_BYTE  : 'b',
-                    ns.EventEntity.EVENT_WORD  : 'h',
-                    ns.EventEntity.EVENT_DWORD : 'i'}
-        val_type = type_map[event.event_type]
-        if val_type == 'a':
-            val_type += str(event.max_data_length)
-        return np.dtype([('timestamp', 'd'), ('value', val_type)])
-
-
-class ConsoleIndicator(ProgressIndicator):
-    def __init__(self):
-        super(ConsoleIndicator, self).__init__()
-        self._size = 60
-        self._last_msg = ""
-
-    def progress(self,  max_value, cur_value):
-        size = self._size
-        prefix = "Converting"
-        x = int(size*cur_value/max_value)
-        msg = "%s [%s%s] %i/%i\r" % (prefix, "#"*x, "." * (size-x),
-                                     cur_value, max_value)
-        self._last_msg = msg
-        sys.stdout.write(msg)
-        sys.stdout.flush()
-
-    def cleanup(self):
-        sys.stdout.write('%s\r' % (' '*len(self._last_msg)))
-        sys.stdout.flush()
-
-
-def main():
-    opts, rem = getopt.getopt(sys.argv[1:], 'o:', ['output=',
-                                                   'version=',
-                                                   ])
-    output = None
-    for opt, arg in opts:
-        if opt in ("-o", "--output"):
-            output = arg
-
-    if len(rem) != 1:
-        print("Wrong number of arguments")
-        return -1
-
-    filename = rem[0]
-    ci = ConsoleIndicator()
-    converter = Converter(filename, output, progress=ci)
-    converter.convert()
-    ci.cleanup()
-    return 0
-
-
-if __name__ == "__main__":
-    res = main()
-    sys.exit(res)
+#!/usr/bin/env python
+from __future__ import print_function
+
+import os
+import sys
+import neuroshare as ns
+import numpy as np
+import time
+import nix
+import h5py
+import argparse
+
+class ProgressIndicator(object):
+    def __init__(self, offset=0):
+        self._cur_value = offset
+        self._max_value = 0
+
+    def setup(self, max_value):
+        self._max_value = max_value
+        self.progress(self._max_value, 0)
+
+    def __add__(self, other):
+        self._cur_value += other
+        self.progress(self._max_value, self._cur_value)
+        return self
+
+    def progress(self, max_value, cur_value):
+        pass
+
+#CLASS DEFINITION FOR NIX(Default) CONVERSION
+class Converter(object):
+    def __init__(self, filepath, output=None, progress=None):
+        if not output:
+            (basefile, ext) = os.path.splitext(filepath)
+            output = "%s.h5" % basefile
+
+        nf = ns.File(filepath)
+        nixF = nix.File.open(output,nix.FileMode.Overwrite)
+        nixB = nixF.create_block("main_block","nix.session")

```

Comment:
`main_block` and `main_section` are not nice names. Not sure what else to use. @jgrewe any ideas? maybe the filename for both?